codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -25,54 +26,67 @@ let make = (
 ) => {
   let numAddresses = staticContracts->Array.length + dynamicContractRegistrations->Array.length
 
-  let partitions = if numAddresses <= maxAddrInPartition {
-    list{
-      FetchState.makeRoot(~endBlock)(
-        ~staticContracts,
-        ~dynamicContractRegistrations,
-        ~startBlock,
-        ~logger,
-        ~isFetchingAtHead=false,
-      ),
+  let newestPartitionIndexRef = ref(None)
+  let partitions = Js.Dict.empty()
+  let setNextPartition = partition => {
+    let nextPartitionIndex = switch newestPartitionIndexRef.contents {
+    | Some(newestPartitionIndex) => newestPartitionIndex + 1
+    | None => 0
     }
+    newestPartitionIndexRef := Some(nextPartitionIndex)
+    partitions->Js.Dict.set(nextPartitionIndex->Int.toString, partition)
+  }
+
+  let getNewestPartitionIndex = () => {
+    switch newestPartitionIndexRef.contents {
+    | Some(newestPartitionIndex) => newestPartitionIndex
+    | None => Js.Exn.raiseError("Unexpected no part")
```

Comment:
Should `newestPartitionIndexRef.contents` maybe always be `int` instead of `option<int>`?